general/getting-started.mdx

```diff
@@ -1,10 +1,149 @@
+
+```bash
+npm install @lemoncode/fonk --save
+```
+
+> **Fonk** is Typescript friendly, typing definitions are already included in the main package.
+
+## Using
+
+Fonk is framework agnostic, let's start by adding some simple code to define a login validation:
+
+```javascript
+const loginRecord = {
+  user: "",
+  password: ""
+};
+```
+
+Let's define the following validation schema for this form:
+
+- User field is mandatory.
+
+- Password field is mandatory.
+
+Now let's create the validation schema:
+
+```javascript
+import { Validators, createFormValidation } from "@lemoncode/fonk";
+
+const validationSchema = {
+  field: {
+    user: [Validators.required.validator],
+    password: [Validators.required.validator]
+  }
+};
+
+const formValidation = createFormValidation(validationSchema);
+```
+
+And let's fire a field validation (usually we will fire this in the *onChange* or *onBlur* event of every HTML field):
+
+```javascript
+formValidation
+  .validateField("user", loginRecord.user)
+  .then(validationResult => {
+    console.log(validationResult);
+  });
+```
+
+In this case we get the following result (validation failed):
+
+```bash
+{
+  succeeded: false,
+  message: "Please fill in this mandatory field.",
+  type: "REQUIRED"
+}
+```
+
+What happens if we feed the login.user field with data?
+
+```javascript
+loginRecord.user = "John";
+```
+
+And we fire again the field valdation
+
+```javascript
+formValidation
+  .validateField("user", loginRecord.user)
+  .then(validationResult => {
+    console.log(validationResult);
+  });
+```
+
+We get the following result (validation succeeded):
+
+```bash
+{
+  succeeded: true,
+  message: "",
+  type: "REQUIRED"
+}
+```
+
+If you want to fire all the form validations in one go, you can execute *validateAll*
+
+```javascript
+formValidation.validateForm(loginRecord).then(validationResult => {
+  console.log(validationResult);
+});
+```
+
+And you will get the result of firing all defined validations in the schema:
+
+```bash
+{
+  succeeded: false,
+  fieldErrors: {
+    user: {
+      succeeded: true,
+      message: "",
+      type: "REQUIRED"
+    },
+    password: {
+      succeeded: false,
+      message: "Please fill in this mandatory field.",
+      type: "REQUIRED"
+    },
+  },
+  recordErrors: {}
+}
+```
+
+## UI Examples
+
+You can play with this example in the following [codesandbox](https://codesandbox.io/s/github/lemoncode/fonk/tree/master/examples/vanilla/js/getting-started).
+
+If you want to check a full example including user interface interaction, check the following examples:
+
+- No framework example: [Examples in Javascript and Typescript](https://github.com/Lemoncode/fonk/tree/master/examples/vanilla)
+
+- React + React Final Forms: [Examples in Javascript and Typescript](https://github.com/Lemoncode/fonk/tree/master/examples/react-final-form)
```

Comment:
Got an aerrr when clicking on the three links (just tried raw view, markdown and clicked on the links)

Reply:
Right now, it's only working the react-final-form one, because I need to create a new task in fonk to add other examples. Can we continue with the doc and then we move to examples?